File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode cur = head;
        while(cur != null) {
            if(cur.next != null && cur.val == cur.next.val) {
                ListNode nxt = cur.next;
                while(nxt.next != null && nxt.val == nxt.next.val) {
                    nxt = nxt.next;
                }
                cur.next = nxt.next;
            }
            cur = cur.next;
        }
        return head;
    }
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur = head
        while cur:
            if cur.next and cur.val == cur.next.val:
                next = cur.next
                while next.next and next.val == next.next.val:
                    next = next.next
                cur.next = next.next
            cur = cur.next
        return head